plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -584,24 +616,62 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
-    public static LongWriteFunction trinoTimestampToOracleTimestampWriteFunction()
+    private static ObjectWriteFunction oracleLongTimestampWriteFunction(TimestampType timestampType)
+    {
+        int roundToPrecision = timestampType.getPrecision();
+        checkArgument(timestampType.getPrecision() > TimestampType.MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
+        checkArgument(
+                6 <= roundToPrecision && roundToPrecision <= 9 && roundToPrecision <= timestampType.getPrecision(),
```

Comment:
I think `oracleLongTimestampWriteFunction` is getting called when precision is greater than `MAX_SHORT_PRECISION`(which is 6) so, the condition should be `6 < roundToPrecision && roundToPrecision <= 9`.

Reply:
can we use `MAX_SHORT_PRECISION` and `MAX_ORACLE_TIMESTAMP_PRECISION` instead 6 and 9 respectively for readability sack?

Reply:
refactored